src/loki/search.cc

```diff
@@ -201,21 +202,9 @@ struct bin_handler_t {
   std::vector<candidate_t> bin_candidates;
   std::unordered_set<uint64_t> correlated_edges;
 
-  // key is the edge id, size_t is the index into the reachability number
-  // which stores the number of nodes you can reach from a given node in the
-  // in the forward direction. TODO: direction is important because it answers
-  // the question, can i get out of here? importantly it currently doesnt answer
-  // the question of whether you can get into a place for that we need the
-  // reverse direction reachability
-  std::vector<unsigned int> reaches;
-  std::unordered_map<uint64_t, size_t> reach_indices;
-
-  // TODO: keep track of a reachability object per node
-  struct reachability_t {
-    size_t round;
-    unsigned int forward_reach;
-    unsigned int backward_reach;
-  };
+  // keep track of edges whose reachability we've already computed
+  // TODO: dont use pointers as keys, its safe for now but fancy caching one day could be bad
+  std::unordered_map<const DirectedEdge*, directed_reach> directed_reaches;
```

Comment:
the main changes in this file rae to delete the previous implementation of reach and replace it with a call to the new one. i wanted to call out that i did this as a shortcut to prove this was working but i feel like we really shouldnt use pointers as keys